### PR TITLE
dotCMS/core#21701 fix When editing a piece of content, update the pag…

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
…e title with the content title

### Proposed Changes
when you open a piece of content, or navigate through the system we change the html page title dynamically based on what you are editing or doing

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: apps/dotcms-ui/src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts [hunk 1]
```diff
@@ -83,6 +85,13 @@ export class DotContentletWrapperComponent {
                 },
                 'edit-contentlet-loaded': (e: CustomEvent) => {
                     this.header = e.detail.data.contentType;
+                    this.titleService.setTitle(
+                        `${
+                            e.detail.data.pageTitle
+                                ? e.detail.data.pageTitle + ' - '
+                                : this.titleService.getTitle()
+                        } ${this.titleService.getTitle().split(' - ')[1]}`
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts:93]: Can't we just hardcode this part `${this.titleService.getTitle().split(' - ')[1]}`? Which I'm not mistaken is "dotCMS Platform"

File: apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts [hunk 2]
```diff
@@ -373,4 +383,17 @@ export class DotNavigationService {
...
+        let title = '';
+        for (let i = 0; i < menu.length; i++) {
+            for (let j = 0; j < menu[i].menuItems.length; j++) {
+                title =
+                    url.indexOf(menu[i].menuItems[j].menuLink) >= 0
+                        ? menu[i].menuItems[j].label
+                        : title;
+            }
+        }
+        return title;
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts:397]: Isn't easier to join all the children in one plain array and then look for the URL to match?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts [hunk 3]
```diff
@@ -67,60 +68,59 @@ function isEditPageFromSiteBrowser(menuId: string, previousUrl: string): boolean
...
+                        if (isActive) {
+                            menus[i].menuItems[k].active = false;
+                        }
+
+                        if (!isActive && menus[i].menuItems[k].id === urlId) {
+                            isActive = true;
+                            menus[i].active = true;
+                            menus[i].isOpen = true;
+                            menus[i].menuItems[k].active = true;
+                        }
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts:116]: not sure to be honest, from line 71 to 123 I did not make a change, it's only formatted by prettier

File: apps/dotcms-ui/src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts (hunk 1 above)
Comment (alfredo-dotcms) [reply apps/dotcms-ui/src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts:93]: are you sure? by doing what I implemented, we keep what ever we have set on the main `index.html` file https://github.com/dotCMS/core-web/blob/397211cc9f4f6b3f657404413f8dc4f72bba5736/apps/dotcms-ui/src/index.html#L10

so any future change to that title is done on a single place

File: apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts (hunk 2 above)
Comment (alfredo-dotcms) [reply apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts:397]: done!

**Review by hmoreras — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**